setup.cfg

```diff
@@ -144,6 +144,7 @@ doctest_norecursedirs =
 doctest_subpackage_requires =
     astropy/io/misc/asdf/* = asdf
     astropy/io/misc/yaml.py = PyYAML
+    astropy/visualization/* = matplotlib
```

Comment:
I don't think everything requires mpl in the submodule, but even then I would be OK with this limitation here.